From fd93cbb85929c15e66b793540e527b866fdb5cdf Mon Sep 17 00:00:00 2001
From: "tao.zeng" <tao.zeng@amlogic.com>
Date: Fri, 14 Mar 2014 15:29:32 +0800
Subject: [PATCH 3743/5965] PD #88585: Build PMU algorithm driver from external
 repository.

Instead of releasing a binary KLM for PMU algorithm driver, switch to build
the proprietary KLM from external repository.
---
 arch/arm/configs/meson6_defconfig | 1 +
 arch/arm/configs/meson8_defconfig | 1 +
 drivers/amlogic/Kconfig           | 1 +
 drivers/amlogic/Makefile          | 4 ++++
 drivers/amlogic/power/Makefile    | 1 -
 5 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm/configs/meson6_defconfig b/arch/arm/configs/meson6_defconfig
index 9fae4834347e..b181992abc73 100755
--- a/arch/arm/configs/meson6_defconfig
+++ b/arch/arm/configs/meson6_defconfig
@@ -371,3 +371,4 @@ CONFIG_AMLOGIC_IONVIDEO=y
 CONFIG_PREEMPT=y
 CONFIG_AML_DVFS=y
 CONFIG_MESON_CS_DCDC_REGULATOR=y
+CONFIG_AML_PMU_ALGORITHM_SUPPORT=m
diff --git a/arch/arm/configs/meson8_defconfig b/arch/arm/configs/meson8_defconfig
index 553420919bfd..3fb0ccd927b0 100755
--- a/arch/arm/configs/meson8_defconfig
+++ b/arch/arm/configs/meson8_defconfig
@@ -358,4 +358,5 @@ CONFIG_CMA=y
 CONFIG_CMA_SIZE_MBYTES=8
 CONFIG_CMA_SIZE_SEL_MBYTES=y
 CONFIG_CMA_ALIGNMENT=8
+CONFIG_AML_PMU_ALGORITHM_SUPPORT=m
 CONFIG_CMA_AREAS=7
diff --git a/drivers/amlogic/Kconfig b/drivers/amlogic/Kconfig
index 50c5f6955178..cbf7c37a18db 100755
--- a/drivers/amlogic/Kconfig
+++ b/drivers/amlogic/Kconfig
@@ -96,4 +96,5 @@ source "drivers/amlogic/ion_dev/Kconfig"
 source "drivers/amlogic/crypto/Kconfig"
 
 source "drivers/amlogic/mhl/Kconfig"
+isource "../hardware/amlogic/pmu/Kconfig"
 endmenu
diff --git a/drivers/amlogic/Makefile b/drivers/amlogic/Makefile
index 4fa32b04a772..19ee38e1bf6b 100755
--- a/drivers/amlogic/Makefile
+++ b/drivers/amlogic/Makefile
@@ -67,6 +67,10 @@ obj-$(CONFIG_AML_RTC)+= rtc/
 
 obj-y += power/
 
+ifneq ($(wildcard $(srctree)/../hardware/amlogic/pmu),)
+    obj-$(CONFIG_AML_PMU_ALGORITHM_SUPPORT) += ../../../hardware/amlogic/pmu/
+endif
+
 obj-y += tvin_v4l2/
 ifneq ($(wildcard $(srctree)/../hardware/tvin),)
 	obj-$(CONFIG_TVIN) += ../../../hardware/tvin/
diff --git a/drivers/amlogic/power/Makefile b/drivers/amlogic/power/Makefile
index 9d5fced42a3e..9d15492a16d0 100755
--- a/drivers/amlogic/power/Makefile
+++ b/drivers/amlogic/power/Makefile
@@ -22,7 +22,6 @@ obj-$(CONFIG_AML_PMU)                       += aml_pmu/
 obj-$(CONFIG_RICOH_PMU)                     += ricoh/
 obj-$(CONFIG_UBOOT_BATTERY_PARAMETERS)      += battery_parameter.o
 obj-$(CONFIG_AMLOGIC_PMU_OF)                += aml_pmu_of_common.o pmu_mutex.o 
-obj-$(CONFIG_AML_PMU_ALGORITHM_SUPPORT)     += aml_pmu_algorithm.o 
 
 endif
 
-- 
2.19.0

